### PR TITLE
improve EdgeRouter reconnect process

### PR DESCRIPTION
**Pull request by ekoby**
* pull uv-mbed update
* add EdgeRouter latency/connect timeouts

### CODE REVIEW

File: includes/ziti/ziti.h [hunk 1]
```diff
@@ -532,6 +532,9 @@ extern int ziti_set_timeout(ziti_context ztx, int timeout);
 ZITI_FUNC
 extern int ziti_shutdown(ziti_context ztx);
 
+ZITI_FUNC
```
Comment (dovholuknf) [inline includes/ziti/ziti.h:535]: need doc pls :)

File: library/channel.c [hunk 2]
```diff
@@ -513,9 +518,20 @@ static void latency_reply_cb(void *ctx, message *reply) {
...
+    else {
         ZITI_LOG(WARN, "invalid latency probe result ct[%d]", reply->header.content);
     }
+    uv_timer_start(&ch->timer, send_latency_probe, LATENCY_INTERVAL, 0);
+}
+
+static void latency_timeout(uv_timer_t *t) {
+    ziti_channel_t *ch = t->data;
+    ziti_channel_remove_waiter(ch, ch->latency_waiter);
+    ch->latency_waiter = NULL;
```
Comment (dovholuknf) [inline library/channel.c:531]: noob question, why not NULL in ziti_channel_remove_waiter? bad form? seems like you could put it close to the waiter free?

File: library/channel.c [hunk 3]
```diff
@@ -560,17 +578,17 @@ static void hello_reply_cb(void *ctx, message *msg) {
...
         r->cb(ch, r->ctx, cb_code);
         free(r);
     }
-    ch->conn_reqs_n = 0;
 
     if (success) {
         // initial latency
         ch->latency = uv_now(ch->loop) - ch->latency;
-        uv_timer_start(&ch->timer, send_latency_probe, 0, 60 * 1000);
+        uv_timer_start(&ch->timer, send_latency_probe, LATENCY_INTERVAL, 0);
```
Comment (dovholuknf) [inline library/channel.c:591]: just checking - i see the params are flopped... was this a bug that was fixed? just wanna make sure it was intentional

File: library/channel.c [hunk 4]
```diff
@@ -617,12 +641,15 @@ static void reconnect_cb(uv_timer_t *t) {
...
         uv_mbed_init(ch->loop, &ch->connection, ch->connection.tls);
-        ch->connection._stream.data = ch;
+        ch->connection.data = ch;
         ZITI_LOG(DEBUG, "connecting ch[%d] to %s:%d", ch->id, ch->host, ch->port);
         int rc = uv_mbed_connect(req, &ch->connection, ch->host, ch->port, on_channel_connect_internal);
         if (rc != 0) {
             on_channel_connect_internal(req, rc);
         }
+        else {
+            uv_timer_start(&ch->timer, connect_timeout, 20 * 1000, 0);
```
Comment (dovholuknf) [inline library/channel.c:651]: hardcoded 20s? maybe a constant declared somewhere else?

**Review by dovholuknf — APPROVED** [review]

File: library/channel.c (hunk 2 above)
Comment (ekoby) [reply library/channel.c:531]: it won't work, pass-by-value

File: includes/ziti/ziti.h (hunk 1 above)
Comment (ekoby) [reply includes/ziti/ziti.h:541]: added

File: library/channel.c (hunk 3 above)
Comment (ekoby) [reply library/channel.c:592]: on purpose

**Pull request merged**